File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/RegEx.java
package com.quicksilver.objectvalidator.runtime;

import java.util.HashSet;

public class RegEx extends Condition {
    private final String pattern;

    public RegEx(boolean reversed, String fieldExpression, String pattern) {
        super(reversed, fieldExpression);
        this.pattern = pattern;
    }

    @Override
    protected boolean isFulfilledBy(Object value, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) {
        return switch (value) {
            case null -> throw new RuntimeException("Null values are not supported for 'regex'.");
            case String s -> s.matches(pattern);
            default -> throw new RuntimeException("Unsupported type for 'regex': " + value.getClass());
        };
    }
}

File: C Sharp/Object Validator/Runtime/RegEx.cs
using System.Text.RegularExpressions;

namespace Quicksilver.ObjectValidator.Runtime;
class RegEx(bool reversed, string? fieldExpression, string pattern) : Condition(reversed, fieldExpression)
{
    private readonly string pattern = pattern;

    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        return value switch {
            null => throw new Exception("Null values are not supported for 'regex'."),
            string s => Regex.IsMatch(s, pattern),
            _ => throw new Exception("Unsupported type for 'regex': " + value.GetType())
        };
    }
}